src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -30,24 +36,66 @@ function HeatmapVis(props: Props): JSX.Element {
     keepAspectRatio = true,
     showGrid = false,
     showLoader = true,
+    abscissas,
+    ordinates,
   } = props;
 
   const { rows, cols } = getDims(dataArray);
   const aspectRatio = keepAspectRatio ? cols / rows : undefined; // width / height <=> cols / rows
 
+  const abscissaValues = abscissas || range(cols + 1);
+  const abscissaToIndex = getValueToIndexScale(abscissaValues);
+  const abscissaDomain = useMemo(() => {
+    return getDataDomain(abscissaValues) ?? DEFAULT_DOMAIN;
```

Comment:
Is the `DEFAULT_DOMAIN` required? If the domain is `undefined`, then the axis won't appear, which gives a good indication to the user that something isn't right. I think the default domain is only useful for the color map, when auto-scale is on.

Reply:
I would agree but this means that to satisfy TS, we would need to accept `undefined` for `domain` all the way down to `Axis`...

Reply:
Right, and we didn't use to because we were always initializing the domain to something... Would it make sense to return an error, then? We could check both domains at once to save time.